doc.go

```diff
@@ -3,14 +3,42 @@ package pty
 
 import (
 	"errors"
-	"os"
+	"io"
 )
 
 // ErrUnsupported is returned if a function is not
 // available on the current platform.
 var ErrUnsupported = errors.New("unsupported")
 
 // Open a pty and its corresponding tty.
-func Open() (pty, tty *os.File, err error) {
+func Open() (Pty, Tty, error) {
```

Comment:
This is a breaking change and will require a new major version. We can do that, but it would be nice to find a way to maintain compatibility with v1.

A common use-case is to store the ptmx and/or pass it around, which currently is *os.File. It would easily be fixed by changing the type to pty.Pty, but still, it is a breaking change.
Another common one is to pass the ptmx as extra param to Cmd, which needs to be *os.File, and that one is a bit more tricky, would require calling os.NewFile() from the fds

Reply:
That's what I am concerned about as noted in this pull request's description, the second method you mentioned above seems won't work for windows platform unless the end user will create the PseudoConsole handle and pass it to the Cmd using startupInfoEx (which requires rewrite of os/exec.Cmd like we did).

Reply:
I could not think of a better solution other than the Pty/Tty interface, we have noted the difference of the underlying struct type for unix and windows platforms, our users should be able to adapt to the new Pty/Tty interface easily (for most broken cases, they just need to cast the returned value to *os.File)

I guess what we need is to note this major api change in the release note.